File: src/main/java/itmo/is/project/repository/trade/TradePolicyRepository.java
package itmo.is.project.repository.trade;

import itmo.is.project.model.trade.TradePolicy;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TradePolicyRepository extends JpaRepository<TradePolicy, Integer> {
}

File: src/main/kotlin/itmo/isproject/repository/trade/TradePolicyRepository.kt
package itmo.isproject.repository.trade

import itmo.isproject.model.trade.TradePolicy
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TradePolicyRepository : JpaRepository<TradePolicy, Int> {

    fun findAllByStationBuysTrue(pageable: Pageable): Page<TradePolicy>

    fun findAllByStationSellsTrue(pageable: Pageable): Page<TradePolicy>
}
